Yank/restraints.py

```diff
@@ -96,330 +107,392 @@ def available_restraint_types():
-            print('bond_parameters: %s' % str(self._bond_parameters))
-            raise(e)
-        return force
+        # Check that restrained atoms are defined.
+        if self.restrained_receptor_atom is None or self.restrained_ligand_atom is None:
+            raise RestraintParameterError('Restraint {}: Undefined restrained '
+                                          'atoms.'.format(self.__class__.__name__))
 
-    def get_restraint_force(self):
-        """
-        Returns a new Force object that imposes the receptor-ligand restraint.
+        # Create restraint force.
+        restraint_force = self._create_restraint_force(self.restrained_receptor_atom,
+                                                       self.restrained_ligand_atom)
 
-        Returns
-        -------
-        force : simtk.openmm.CustomBondForce
-           The created restraint force.
+        # Set periodic conditions on the force if necessary.
+        restraint_force.setUsesPeriodicBoundaryConditions(thermodynamic_state.is_periodic)
 
-        """
-        return self._create_restraint_force(self._restrained_receptor_atom, self._restrained_ligand_atom)
+        # Get a copy of the system of the ThermodynamicState, modify it and set it back.
+        system = thermodynamic_state.system
+        system.addForce(restraint_force)
+        thermodynamic_state.system = system
 
-    def get_standard_state_correction(self):
-        """
-        Return the standard state correction.
+    def get_standard_state_correction(self, thermodynamic_state):
+        """Return the standard state correction.
+
+        Parameters
+        ----------
+        thermodynamic_state : openmmtools.states.ThermodynamicState
+            The thermodynamic state.
 
         Returns
         -------
         correction : float
-           The standard-state correction, in kT
+           The unit-less standard-state correction, in kT (at the
+           temperature of the given thermodynamic state).
 
         """
-        initial_time = time.time()
+        benchmark_id = 'Restraint {}: Computing standard state correction'.format(self.__class__.__name__)
+        timer = mmtools.utils.Timer()
+        timer.start(benchmark_id)
 
         r_min = 0 * unit.nanometers
-        r_max = 100 * unit.nanometers # TODO: Use maximum distance between atoms?
+        r_max = 100 * unit.nanometers  # TODO: Use maximum distance between atoms?
 
         # Create a System object containing two particles connected by the reference force
         system = openmm.System()
         system.addParticle(1.0 * unit.amu)
         system.addParticle(1.0 * unit.amu)
         force = self._create_restraint_force(0, 1)
         # Disable the PBC if on for this approximation of the analytical solution
-        try:  # This was added in OpenMM 7.1
-            force.setUsesPeriodicBoundaryConditions(False)
-        except AttributeError:
-            pass
+        force.setUsesPeriodicBoundaryConditions(False)
```

Comment:
Right! I forgot about that. I set it back to `openmm>=7.0.1` few months ago to make it work on the cluster. I'll revert it in this PR.